app/bundles/FormBundle/Model/SubmissionModel.php

```diff
@@ -214,6 +214,12 @@ public function saveSubmission($post, $server, Form $form, Request $request = nu
             }
 
             if ($f->isRequired() && empty($value)) {
+
+                //field is required, but hidden from form because of 'ShowWhenValueExists'
+                if( null !== $f->getShowWhenValueExists() && $f->getShowWhenValueExists() === false ){
```

Comment:
Shouldn't this line be just:
```
if ($f->getShowWhenValueExists() === false ) {
```
because if something `=== false`, the `!== null` part must be true as well, mustn't it?